components/constellation/constellation.rs

```diff
@@ -579,31 +584,63 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         namespace_id
     }
 
+    /// The registered domain name (aka eTLD+1) for a URL.
+    /// Returns None if the URL has no host name.
+    /// Returns the registered suffix for the host name if it is a domain.
+    /// Leaves the host name alone if it is an IP address.
+    fn reg_host<'a>(&self, url: &'a ServoUrl) -> Option<&'a str> {
+        url.domain().map(reg_suffix).or(url.host_str())
+    }
+
     /// Helper function for creating a pipeline
     fn new_pipeline(&mut self,
                     pipeline_id: PipelineId,
                     frame_id: FrameId,
                     parent_info: Option<(PipelineId, FrameType)>,
-                    old_pipeline_id: Option<PipelineId>,
                     initial_window_size: Option<TypedSize2D<f32, PagePx>>,
-                    script_channel: Option<Rc<ScriptChan>>,
                     load_data: LoadData,
+                    sandbox: IFrameSandboxState,
                     is_private: bool) {
         if self.shutting_down { return; }
 
+        // TODO: think about the case where the child pipeline is created
+        // before the parent is part of the frame tree.
```

Comment:
I made this comment into a question. You're right, I don't think this can happen, but I'm not sure we should be relying on that.